### ISSUE TITLE
system/ThirdParty laminas-escaper version

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#8029]
https://packagist.org/packages/laminas/laminas-escaper
2.12.0 supports PHP 7.4 to 8.2.
2.13.0 supports PHP 8.1 to 8.3.

CI 4.4 supports PHP 7.4, so we need to include escaper 2.12.0, but if we use PHP 8.1 or later,
when we run `composer update`, the `system/ThirdParty/Escaper/Escaper.php` file is updated by 2.13.0.
It is simply bothersome to have to delete the change every time.

If CI 4.5 supports PHP 8.0, the same thing will happen.

Is there any good solution?

**Comment by paulbalandan** [comment]
How about `~2.12`?

**Comment by kenjis** [comment]
`~2.12.0` is equivalent to `>=2.12.0 <2.13.0`.
If a dev wants to use 2.13 with PHP 8.1, they can't.

But the changes from 2.12 to 2.13 are trivial. So it may be no problem.
https://github.com/laminas/laminas-escaper/releases/tag/2.13.0

**Comment by kenjis** [comment]
No, it is not good. If we use `~2.12.0`, we cannot use PHP 8.3.

```console
$ composer update
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires laminas/laminas-escaper ~2.12.0 -> satisfiable by laminas/laminas-escaper[2.12.0].
    - laminas/laminas-escaper 2.12.0 requires php ^7.4 || ~8.0.0 || ~8.1.0 || ~8.2.0 -> your php version (8.3.0-dev) does not satisfy that requirement.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```

**Comment by ddevsr** [comment]
Why not `~2.12.0 | ~2.13.0`?

**Comment by kenjis** [comment]
CI4 includes laminas-escaper in `system/ThirdParty`.
Because CI4 (Zip installation) runs without Composer.
And the source code in `system/ThirdParty` is updated automatically when we run `composer update` on this repository.

If I use PHP 8.1, 2.13.0 is installed when we set `~2.12.0 | ~2.13.0`.
But for CI 4.4, we must include 2.12.0, because 4.4 supports PHP 7.4.

**Comment by kenjis** [comment]
This problem is not resolved in `4.4`. 
However, it will not occur in `4.5` because the version will be raised to 2.13.
